Source/carthage/Formatting.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  Formatting.swift
+//  Carthage
+//
+//  Created by J.D. Healy on 1/29/15.
+//  Copyright (c) 2015 Carthage. All rights reserved.
+//
+
+import Commandant
+import Foundation
+import LlamaKit
+import PrettyColors
+
+/// Wraps a string with terminal colors and formatting or passes it through, depending on `colorful`.
+private func wrap(colorful: Bool)(wrap: Color.Wrap)(string: String) -> String {
+	return colorful ? wrap.wrap(string) : string
+}
+
+/// Argument for whether to color and format terminal output.
+public enum ColorArgument: String, ArgumentType, Printable {
+	case Auto = "auto"
+	case Never = "never"
+	case Always = "always"
+	
+	/// Whether to color and format.
+	public var isColorful: Bool {
+		switch self {
+		case .Always:
+			return true
+		case .Never:
+			return false
+		case .Auto:
+			return Terminal.isTTY && !Terminal.isDumb
```

Comment:
I was wondering if we’d want to check whether `TERM` contains the string `color`, but it’s probably not crucial for this PR.

There’s also apparently [`tput colors`](http://blog.sanctum.geek.nz/terminal-colour-tolerance/) if we ever wanted to get fancy and use more/fewer colours depending on the `TERM`’s support for them. That _definitely_ doesn’t need to be addressed in this PR, however.